operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -359,6 +359,18 @@ func (r *RedpandaReconciler) reconcileDefluxed(ctx context.Context, rp *redpanda
 	// DeepCopy values to prevent any accidental mutations that may occur
 	// within the chart itself.
 	values := rp.Spec.ClusterSpec.DeepCopy()
+	if c := apimeta.FindStatusCondition(rp.Status.Conditions, redpandav1alpha2.ClusterConfigSynced); c != nil && c.Status == metav1.ConditionTrue {
+		if values.Statefulset == nil {
+			values.Statefulset = &redpandav1alpha2.Statefulset{}
+		}
+		if values.Statefulset.PodTemplate == nil {
+			values.Statefulset.PodTemplate = &redpandav1alpha2.PodTemplate{}
+		}
+		if values.Statefulset.PodTemplate.Annotations == nil {
+			values.Statefulset.PodTemplate.Annotations = map[string]string{}
+		}
+		values.Statefulset.PodTemplate.Annotations["redpanda.operator.redpanda.com/cluster-config-version"] = c.Message
```

Comment:
NIT:

We are using `operator.redpanda.com` not redpanda.operator.redpanda.com` in finalizers and other annotations
```suggestion
		values.Statefulset.PodTemplate.Annotations["operator.redpanda.com/cluster-config-version"] = c.Message
```

It would be good to hoist this as const.